### DISCUSSION TITLE
Hidden variable support

### DISCUSSION THREAD

**Discussion by kodecrash** [Budibase/budibase#6425] [Help]
I want to implement custom logic in Embed component.

I am writing JavaScript code based on my data provider input and I need to store hidden variable in the page. So that I can use that variable for some calculation.

How to add hidden variable. I tried adding number field **"currentStep"** but I am unable to get this variable in binding.
![image](https://user-images.githubusercontent.com/8296994/175011830-c0c6b0e9-6492-4c80-bb7c-0e8751baf845.png)

How to get that field here in the handlerbar or javascript code.

![image](https://user-images.githubusercontent.com/8296994/175011991-8f166f2c-531b-42db-8fa4-fdc0277913f9.png)

**Comment by poirazis** [comment]
Hey @kodecrash 
have you tried using App State ? https://docs.budibase.com/docs/app-state

**Reply by kodecrash** [reply]
How to use app state. I am not getting it just reading the documentation.
I want to set a variable in APP state on page load.
Could you please provide any example here?
